src/core/castImage.ts

```diff
@@ -0,0 +1,158 @@
+import PixelTypes from './PixelTypes.js'
+import IntTypes from './IntTypes.js'
+import FloatTypes from './FloatTypes.js'
+
+/**
+ * Cast an image to another PixelType or ComponentType
+ *
+ * @param {Image} image - The input image
+ * @param {CastImageOptions} options - specify the componentType and/or pixelType of the output
+ */
+function castImage (inputImage: Image, options: CastImageOptions): Image {
+  const outputImageType = { ...inputImage.imageType }
+
+  if (typeof options.pixelType !== 'undefined') {
+    outputImageType.pixelType = options.pixelType
+    if (options.pixelType === PixelTypes.Scalar && outputImageType.components !== 1) {
+      throw new Error('Cannot cast multi-component image to a scalar image')
+    }
+  }
+  if (typeof options.componentType !== 'undefined' && options.componentType !== inputImage.imageType.componentType) {
+    outputImageType.componentType = options.componentType
+  }
+
+  const outputImage = new Image(outputImageType)
+
+  outputImage.name = inputImage.name
+  outputImage.origin = Array.from(inputImage.origin)
+  outputImage.spacing = Array.from(inputImage.spacing)
+  outputImage.direction = inputImage.direction.slice()
+  outputImage.size = Array.from(inputImage.size)
+
+  if (inputImage.data !== null) {
+    if (typeof options.componentType !== 'undefined' && options.componentType !== inputImage.imageType.componentType) {
+      switch (inputImage.imageType.componentType) {
+        case IntTypes.UInt8:
+        case IntTypes.Int8:
+        case IntTypes.UInt16:
+        case IntTypes.Int16:
+        case IntTypes.UInt32:
+        case IntTypes.Int32:
+        case FloatTypes.Float32:
+        case FloatTypes.Float64:
+          switch (outputImage.imageType.componentType) {
+            case IntTypes.UInt8:
+              outputImage.data = new Uint8Array(inputImage.data)
+              break
+            case IntTypes.Int8:
+              outputImage.data = new Int8Array(inputImage.data)
+              break
+            case IntTypes.UInt16:
+              outputImage.data = new Uint16Array(inputImage.data)
+              break
+            case IntTypes.Int16:
+              outputImage.data = new Int16Array(inputImage.data)
+              break
+            case IntTypes.UInt32:
+              outputImage.data = new Uint32Array(inputImage.data)
+              break
+            case IntTypes.Int32:
+              outputImage.data = new Int32Array(inputImage.data)
+              break
+            case FloatTypes.Float32:
+              outputImage.data = new Float32Array(inputImage.data)
+              break
+            case FloatTypes.Float64:
+              outputImage.data = new Float64Array(inputImage.data)
+              break
+            case IntTypes.UInt64:
+              outputImage.data = new BigUint64Array(inputImage.data.length)
+              for (let idx = 0; idx < outputImage.data.length; idx++) {
+                outputImage.data[idx] = BigInt(inputImage.data[idx])
```

Comment:
Should we use `BigInt.asUintN(64, num)` here?

Reply:
Yes!